### PR TITLE
Support Zarr format from the CLI

### PR DESCRIPTION
**Pull request by carlosgjs**
- Adds a `--format zarr|asdf` argument to the CLI
- Additional timing measurement and logging
- Update the test matrix to:
    - Include for format (zarr vs asdf)
    - Be more sparse. All types of normalization and stacking are tested, but not all combinations, which is overkill

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -42,8 +42,18 @@ jobs:
...
+        method: [linear]
+        freq_norm: [rma]
+        format: [asdf]
+        include:
+        - method: [robust, nroot, selective, auto_covariance, pws]
+          python_version: ['3.8', '3.10']
+          freq_norm: [no]
+          format: [zarr]
+        - method: [robust]
+          python_version: ['3.8', '3.10']
```
Comment (lsetiawan) [inline .github/workflows/test.yaml:54]: Any reason why Python 3.9 is not included in the testing?

File: .github/workflows/precommit.yaml [hunk 2]
```diff
@@ -16,6 +16,6 @@ jobs:
       with:
         python-version: 3.8
     - name: Install pre-commit
-      run: pip install pre-commit
+      run: pip install pre-commit==3.3
```
Comment (lsetiawan) [inline .github/workflows/precommit.yaml:19]: Maybe make this pinning same as in `pyproject.toml`?

File: src/noisepy/seis/main.py [hunk 3]
```diff
@@ -37,6 +38,11 @@ class Command(Enum):
     ALL = 4
 
 
+class DataFormat(Enum):
+    ZARR = 1
+    ASDF = 2
```
Comment (lsetiawan) [inline src/noisepy/seis/main.py:43]: Are you actually using the values of this enum anywhere? Would it be better to be the lower name values instead? Seems like that's the only thing you're using it for, might save some of the `.lower` computation.

File: .github/workflows/test.yaml (hunk 1 above)
Comment (carlosgjs) [reply .github/workflows/test.yaml]: Just to keep the matrix smaller since I was doing the full matrix testing. I opted for testing the min and max of supported versions. But actually now that I discovered the `include` argument I can add `3.9` for a single configuration as a good sanity check.

File: src/noisepy/seis/main.py (hunk 3 above)
Comment (carlosgjs) [reply src/noisepy/seis/main.py:43]: It's a habit since an int comparison is generally much faster than a string comparison and has fewer gotchas (like casing and culture). But I agree in this case the string value makes things simpler and won't make a difference perf-wise. I'll update ...

File: .github/workflows/precommit.yaml (hunk 2 above)
Comment (carlosgjs) [reply .github/workflows/precommit.yaml]: yep, already done

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -42,8 +42,18 @@ jobs:
       fail-fast: true
       matrix:
         python_version: ['3.8', '3.10']
```
Comment (lsetiawan) [inline .github/workflows/test.yaml:44]: Does this software not work with python 3.9? I'm curious why that's not included.

File: tests/test_cli.sh [hunk 5]
```diff
@@ -1,14 +1,42 @@
...
+       echo "Missing FORMAT argument (zarr or asdf)"
+       exit 1
+fi
+echo "FORMAT is $FORMAT"
+LOGFILE="$HOME/logs/log_${FORMAT}_$(date -j +'%Y%m%d_%H%M%S').txt"
+STATIONS=ARV,BAK
+START=2019-02-01T00:00:00
+END=2019-02-01T01:00:00
+INC_HOURS=1
+
```
Comment (carlosgjs) [inline tests/test_cli.sh:16]: Results for this "bigger test":
image
This is running on my Mac, so local SSD I/O. All times in seconds.
TL; DR: Zarr is slower for writing, but faster for reading. But also note that there is not a huge difference in the overall time. However, this may change with: a) parallelization of stacking and b) cloud storage

File: .github/workflows/test.yaml (hunk 4 above)
Comment (carlosgjs) [reply .github/workflows/test.yaml:44]: same as above

File: src/noisepy/seis/S2_stacking.py [hunk 6]
```diff
@@ -250,6 +253,7 @@ def write_stacks(comp: str, tparameters: Dict[str, Any], stacks: List[Tuple[Stac
                     ]
                     write_stacks(comp, tparameters, stacks)
         tlog.log(f"stack/rotate all station pairs {pairs_all[ipair]}", t_load)
+        tlog.log_raw("store.append", t_append)
```
Comment (carlosgjs) [inline src/noisepy/seis/S2_stacking.py:256]: @niyiyu

File: src/noisepy/seis/S1_fft_cc_MPI.py [hunk 7]
```diff
@@ -142,11 +144,15 @@ def init() -> List:
...
 
+        t_append = 0
         for t in as_completed(tasks):
             # Use as_completed so we can start saving results to the store
             # while other computations are still running
             src_chan, rec_chan, parameters, corr = t.result()
+            t_start = time.time()
             cc_store.append(ts, src_chan, rec_chan, parameters, corr)
+            t_append += time.time() - t_start
+        tlog.log_raw("Append to store", t_append)
```
Comment (carlosgjs) [inline src/noisepy/seis/S1_fft_cc_MPI.py:155]: @niyiyu

File: .github/workflows/test.yaml (hunk 1 above)
Comment (carlosgjs) [reply .github/workflows/test.yaml]: Added one test with 3.9

**Review by niyiyu — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 8]
```diff
@@ -12,6 +12,7 @@ repos:
       - id: check-json
       - id: check-yaml
       - id: pretty-format-json
+        files: \.(json|json5|js)$
```
Comment (lsetiawan) [inline .pre-commit-config.yaml:15]: Maybe instead of specifying what to include, exclude the `.ipynb` file format since we don't want that to be prettified anyways.
